src/software/thunderscope/robot_communication.py

```diff
@@ -322,7 +324,13 @@ def __exit__(self, type, value, traceback) -> None:
 
         """
         self.running = False
-        self.receive_ssl_wrapper.close()
+
+        if self.receive_ssl_wrapper:
+            self.receive_ssl_wrapper.close()
+
+        if self.receive_ssl_wrapper:
+            self.receive_ssl_referee_proto.close()
```

Comment:
oh William is right, I misread this line
```suggestion
        if self.receive_ssl_referee_proto:
            self.receive_ssl_referee_proto.close()
```